stl/inc/xutility

```diff
@@ -1513,26 +1513,30 @@ using _Unwrap_enum_t = typename _Unwrap_enum<_Elem>::type;
 #define _DEBUG_ORDER_SET_UNWRAPPED(otherIter, first, last, pred) \
     _STD _Debug_order_set_unchecked<otherIter>(first, last, pred)
 
-template <class _Pr, class _Ty1, class _Ty2,
-    enable_if_t<is_same_v<_Remove_cvref_t<_Ty1>, _Remove_cvref_t<_Ty2>>, int> = 0>
+template <class _Pr, class _Ty1, class _Ty2>
+constexpr bool _Enable_debug_lt_pred_order_check = is_same_v<_Remove_cvref_t<_Ty1>, _Remove_cvref_t<_Ty2>>;
+
+template <class _Pr, class _Ty1, class _Ty2, bool _Order_check = _Enable_debug_lt_pred_order_check<_Pr, _Ty1, _Ty2>>
+constexpr bool _Debug_lt_pred_order_check_noexcept = noexcept(declval<_Pr>()(declval<_Ty1>(), declval<_Ty2>()));

```

Comment:
The `declval`s are changing semantics. This function is unusual - it has a perfect forwarding signature, but invokes `_Pred(_Left, _Right)` with lvalues. For the SFINAE condition being changed to `if constexpr`, `_Remove_cvref_t` eliminates the trace of perfect forwarding, but the `noexcept` condition isn't. We should use lvalue references here, to match the invocations.